extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -0,0 +1,499 @@
+          // non-deterministic point of time. Cleaners are introduced to mitigate possible errors, not to escalate them.
+          if (calledManually && lastCacheState instanceof VersionedCache) {
+            ((VersionedCache) lastCacheState).cacheHandler.close();
+          }
+        }
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    private void logExecutionError()
+    {
+      if (updaterFuture.isDone()) {
+        try {
+          updaterFuture.get();
+        }
+        catch (ExecutionException ee) {
+          log.error(ee.getCause(), "Error in %s", this);
+        }
+        catch (CancellationException ce) {
+          log.error(ce, "Future for %s has already been cancelled", this);
+        }
+        catch (InterruptedException ie) {
+          Thread.currentThread().interrupt();
+          throw new RuntimeException(ie);
+        }
+      }
+    }
+
+    @Override
+    public String toString()
+    {
+      return asString;
+    }
+  }
+
+  public interface CacheState
+  {}
+
+  public enum NoCache implements CacheState
+  {
+    CACHE_NOT_INITIALIZED,
+    ENTRY_CLOSED
+  }
+
+  public final class VersionedCache implements CacheState, AutoCloseable
+  {
+    final String entryId;
+    final CacheHandler cacheHandler;
+    final String version;
+
+    private VersionedCache(String entryId, CacheHandler cacheHandler, String version)
+    {
+      this.entryId = entryId;
+      this.cacheHandler = cacheHandler;
+      this.version = version;
+    }
+
+    public Map<String, String> getCache()
+    {
+      return cacheHandler.getCache();
+    }
+
+    public String getVersion()
+    {
+      return version;
+    }
+
+    @Override
+    public void close()
+    {
+      log.debug("Closing version [%s] of %s", version, entryId);
+      cacheHandler.close();
+    }
+  }
+
+  private final Map<Class<? extends ExtractionNamespace>, ExtractionNamespaceCacheFactory<?>> namespacePopulatorMap;
+  private final NamespaceExtractionCacheManager cacheManager;
+  private final AtomicLong updatesStarted = new AtomicLong(0);
+  private final AtomicInteger activeEntries = new AtomicInteger();
+
+  @Inject
+  public CacheScheduler(
+      final ServiceEmitter serviceEmitter,
+      final Map<Class<? extends ExtractionNamespace>, ExtractionNamespaceCacheFactory<?>> namespacePopulatorMap,
+      NamespaceExtractionCacheManager cacheManager
+  )
+  {
+    this.namespacePopulatorMap = namespacePopulatorMap;
+    this.cacheManager = cacheManager;
+    cacheManager.scheduledExecutorService().scheduleAtFixedRate(
+        new Runnable()
+        {
+          long priorUpdatesStarted = 0L;
+
+          @Override
+          public void run()
+          {
+            try {
+              final long tasks = updatesStarted.get();
+              serviceEmitter.emit(
+                  ServiceMetricEvent.builder()
+                                    .build("namespace/deltaTasksStarted", tasks - priorUpdatesStarted)
```

Comment:
We were emitting `namespace/deltaTasksStarted` before. Should I changed it, or postpone to 0.10.0? Or 0.9.3 is in fact 0.10.0?